internal/manifests/targetallocator/container.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package targetallocator
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"github.com/aws/amazon-cloudwatch-agent-operator/apis/v1alpha1"
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/config"
+	"github.com/aws/amazon-cloudwatch-agent-operator/internal/naming"
+)
+
+// Container builds a container for the given TargetAllocator.
+func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.AmazonCloudWatchAgent) corev1.Container {
+	image := otelcol.Spec.TargetAllocator.Image
+	if len(image) == 0 {
+		image = cfg.TargetAllocatorImage()
+	}
+
+	ports := make([]corev1.ContainerPort, 0)
+	ports = append(ports, corev1.ContainerPort{
+		Name:          "http",
+		ContainerPort: 8080,
```

Comment:
make this a variable and re-use it in `internal/manifest/targetallocator/service.go` . This makes it easier to maintain it.